### ISSUE TITLE
Selector.GetHandle is optimized by the AOT compiler, and the current implementation only supports IntPtr, so we can't switch to NativeHandle quite yet (the AOT compiler crashes).

### ISSUE THREAD

**Issue by spouliot** [xamarin/xamarin-macios#14139]
I noticed this one, being `IntPtr` and not `NativeHandle`

Comment from https://github.com/xamarin/xamarin-macios/blob/f36606e654d6e6f5e78411b8524e160691827e33/src/ObjCRuntime/Selector.cs#L143

This was added inside (very large) PR https://github.com/xamarin/xamarin-macios/pull/13356 but I could not find an open (or closed) issue for it

Close if duplicate :)

**Comment by rolfbjarne** [comment]
Ref: https://github.com/dotnet/runtime/pull/61987

I don't think we'll have time to look at this for .NET 6 (given that this will need runtime changes as well), so I'm moving it to .NET 7

**Comment by spouliot** [comment]
@rolfbjarne hmmm... no :-) The `IntPtr` is not the argument (it's a `string`) but the return type. 

```csharp
public extern static /* SEL */ IntPtr GetHandle (/* const char* */ string name);
```

You expose still expose a new one with a different name, but it's uglier.

**Comment by rolfbjarne** [comment]
> @rolfbjarne hmmm... no :-) The `IntPtr` is not the argument (it's a `string`) but the return type.

You're right, I thought the same thing while implementing NativeHandle, then after quite a while came to the that conclusion, and promptly forgot about it again 🤦

**Comment by rolfbjarne** [comment]
Closing, because this would be a breaking change to fix.
